engine-tests/src/test/java/org/terasology/world/chunks/pipeline/ChunkProcessingPipelineTest.java

```diff
@@ -0,0 +1,265 @@
+import org.joml.Vector3i;
+import org.joml.Vector3ic;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.TestInstance;
+import org.terasology.TerasologyTestingEnvironment;
+import org.terasology.assets.management.AssetManager;
+import org.terasology.math.JomlUtil;
+import org.terasology.registry.CoreRegistry;
+import org.terasology.world.block.BlockManager;
+import org.terasology.world.block.internal.BlockManagerImpl;
+import org.terasology.world.block.tiles.NullWorldAtlas;
+import org.terasology.world.chunks.Chunk;
+import org.terasology.world.chunks.blockdata.ExtraBlockDataManager;
+import org.terasology.world.chunks.internal.ChunkImpl;
+import org.terasology.world.chunks.pipeline.stages.ChunkTaskProvider;
+
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Random;
+import java.util.Set;
+import java.util.concurrent.CancellationException;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.Future;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+import java.util.concurrent.atomic.AtomicReference;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+@TestInstance(TestInstance.Lifecycle.PER_METHOD)
+class ChunkProcessingPipelineTest extends TerasologyTestingEnvironment {
+
+    private final BlockManager blockManager = new BlockManagerImpl(new NullWorldAtlas(),
+            CoreRegistry.get(AssetManager.class));
+    private final ExtraBlockDataManager extraDataManager = new ExtraBlockDataManager();
+    private ChunkProcessingPipeline pipeline;
+
+    @Test
+    void simpleProcessingSuccess() throws ExecutionException, InterruptedException, TimeoutException {
+        pipeline = new ChunkProcessingPipeline((p) -> null, (o1, o2) -> 0);
+
+        org.terasology.math.geom.Vector3i chunkPos = new org.terasology.math.geom.Vector3i(0, 0, 0);
+        Chunk chunk = createChunkAt(chunkPos);
+
+        pipeline.addStage(ChunkTaskProvider.create("dummy task", (c) -> c));
+
+        Future<Chunk> chunkFuture = pipeline.invokeGeneratorTask(new Vector3i(0, 0, 0), () -> chunk);
+        Chunk chunkAfterProcessing = chunkFuture.get(1, TimeUnit.SECONDS);
+
+        Assertions.assertEquals(chunkAfterProcessing.getPosition(new Vector3i()), chunk.getPosition(new Vector3i()),
+                "Chunk after processing must have equals position, probably pipeline lost you chunk");
+    }
+
+    @Test
+    void simpleStopProcessingSuccess() {
+        pipeline = new ChunkProcessingPipeline((p) -> null, (o1, o2) -> 0);
+
+        Vector3i position = new Vector3i(0, 0, 0);
+        Chunk chunk = createChunkAt(position);
+
+
+        pipeline.addStage(ChunkTaskProvider.create("dummy long executing task", (c) -> {
+            try {
+                Thread.sleep(1_000);
```

Comment:
are these real `sleep`s in the tests, not using some mock clock that doesn't actually use real time?

I think it's worth taking another look at this to see if we can come up with something that is faster but still reliable.